cmd/juju/cloud/listcredentials.go

```diff
@@ -156,35 +156,64 @@ func (c *listCredentialsCommand) personalClouds() (map[string]jujucloud.Cloud, e
 	return c.personalCloudsFunc()
 }
 
-func (c *listCredentialsCommand) Run(ctxt *cmd.Context) error {
-	var credentials map[string]jujucloud.CloudCredential
-	credentials, err := c.store.AllCredentials()
-	if err != nil && !errors.IsNotFound(err) {
-		return err
-	}
-	if c.cloudName != "" {
-		for cloudName := range credentials {
-			if cloudName != c.cloudName {
-				delete(credentials, cloudName)
-			}
-		}
+func (c *listCredentialsCommand) publicClouds() (map[string]jujucloud.Cloud, error) {
+	publicClouds, _, err := jujucloud.PublicCloudMetadata(jujucloud.JujuPublicCloudsPath())
+	if err != nil {
+		return nil, err
 	}
+	return publicClouds, nil
+}
 
-	// Find local cloud names.
-	personalClouds, err := c.personalClouds()
+func (c *listCredentialsCommand) builtInClouds() (map[string]jujucloud.Cloud, error) {
+	builtin, err := common.BuiltInClouds()
 	if err != nil {
-		return err
+		return nil, errors.Trace(err)
 	}
-	var personalCloudNames []string
-	for name := range personalClouds {
-		personalCloudNames = append(personalCloudNames, name)
+	return builtin, nil
+}
+
+func (c *listCredentialsCommand) Run(ctxt *cmd.Context) error {
+	var cloudNames []string
+
+	if c.cloudName != "" {
+		cloudNames = []string{c.cloudName}
+	} else {
+		// Find cloud names.
```

Comment:
Consider breaking out the gathering of cloud names to a helper func
Then no real need to bother with the older "one line" helper funcs for builtinClouds(), publicClouds() etc